controllers/dns_helper.go

```diff
@@ -208,7 +208,6 @@ func (dh *dnsHelper) getLoadBalancedEndpoints(mcgTarget *multicluster.GatewayTar
 	}
 
 	var endpoint *externaldns.Endpoint
-	var defaultEndpoint *externaldns.Endpoint
```

Comment:
@makslion fyi, i removed this as it's not needed any longer. Also added some more relevant comments below.